### PR TITLE
SONAR-7364 remove SQALE characteristics from scanner API

### PR DESCRIPTION
**Pull request by simonbrandhof**
Characteristics have been deprecated in v4.3.

### CODE REVIEW

File: sonar-plugin-api/src/main/java/org/sonar/api/batch/debt/internal/DefaultDebtModel.java [hunk 1]
```diff
@@ -19,89 +19,37 @@
...
-  }
-
   @Override
   public List<DebtCharacteristic> characteristics() {
-    return newArrayList(characteristicsByKey.get(null));
+    return Collections.emptyList();
   }
 
   @Override
   public List<DebtCharacteristic> subCharacteristics(String characteristicKey) {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/batch/debt/internal/DefaultDebtModel.java]: MINOR Remove this use of "DebtCharacteristic"; it is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: sonar-plugin-api/src/main/java/org/sonar/api/batch/debt/internal/DefaultDebtModel.java [hunk 2]
```diff
@@ -19,89 +19,37 @@
...
 
   @Override
   @CheckForNull
   public DebtCharacteristic characteristicByKey(final String key) {
-    return Iterables.find(characteristicsByKey.values(), new MatchDebtCharacteristicKey(key), null);
+    return null;
   }
 
   @CheckForNull
   public DebtCharacteristic characteristicById(int id) {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/batch/debt/internal/DefaultDebtModel.java]: MINOR Remove this use of "DebtCharacteristic"; it is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: sonar-plugin-api/src/main/java/org/sonar/api/batch/debt/internal/DefaultDebtModel.java [hunk 3]
```diff
@@ -19,89 +19,37 @@
...
   }
 
   @Override
   public List<DebtCharacteristic> subCharacteristics(String characteristicKey) {
-    return newArrayList(characteristicsByKey.get(characteristicKey));
+    return Collections.emptyList();
   }
 
   @Override
   public List<DebtCharacteristic> allCharacteristics() {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/batch/debt/internal/DefaultDebtModel.java]: MINOR Remove this use of "DebtCharacteristic"; it is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: sonar-plugin-api/src/main/java/org/sonar/api/batch/debt/internal/DefaultDebtModel.java [hunk 4]
```diff
@@ -19,89 +19,37 @@
...
-    return newArrayList(characteristicsByKey.values());
+    return Collections.emptyList();
   }
 
   @Override
   @CheckForNull
   public DebtCharacteristic characteristicByKey(final String key) {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/batch/debt/internal/DefaultDebtModel.java]: MINOR Remove this use of "DebtCharacteristic"; it is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: sonar-plugin-api/src/main/java/org/sonar/api/batch/debt/internal/DefaultDebtModel.java [hunk 5]
```diff
@@ -19,89 +19,37 @@
...
-  }
-
-
-  public DefaultDebtModel addSubCharacteristic(DebtCharacteristic subCharacteristic, String characteristicKey) {
-    characteristicsByKey.put(characteristicKey, subCharacteristic);
-    return this;
-  }
-
   @Override
   public List<DebtCharacteristic> characteristics() {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/batch/debt/internal/DefaultDebtModel.java]: MINOR Remove this use of "DebtCharacteristic"; it is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: sonar-plugin-api/src/main/java/org/sonar/api/batch/debt/DebtCharacteristic.java [hunk 6]
```diff
@@ -23,7 +23,9 @@
 
 /**
  * @since 4.3
+ * @deprecated since 5.1 debt model will soon be unavailable on batch side
  */
+@Deprecated
 public interface DebtCharacteristic {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/batch/debt/DebtCharacteristic.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 6 issues:
* MINOR 5 minor
* INFO 1 info

Watch the comments in this conversation to review them.

File: sonar-plugin-api/src/main/java/org/sonar/api/batch/debt/DebtCharacteristic.java [hunk 7]
```diff
@@ -23,7 +23,9 @@
 
 /**
  * @since 4.3
+ * @deprecated since 5.1 debt model will soon be unavailable on batch side
```
Comment (henryju) [inline sonar-plugin-api/src/main/java/org/sonar/api/batch/debt/DebtCharacteristic.java]: :)

**Comment by henryju** [comment]
LGTM

**Pull request closed without merging**